libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -198,7 +199,7 @@ void AP_ADSB::init(void)
     if (detected_num_instances == 0) {
         for (uint8_t i=0; i<ADSB_MAX_INSTANCES; i++) {
             detect_instance(i);
-            if (_backend[i] != nullptr) {
+            if (_backend[i] != nullptr && _backend[i]->init()) {
```

Comment:
Perhaps we should init straight after detection and delete the driver if it fails - it doesn't look like we'll ever do anything useful with it.

Reply:
I've reworked this to:
```
    if (detected_num_instances == 0) {
        for (uint8_t i=0; i<ADSB_MAX_INSTANCES; i++) {
            detect_instance(i);
            if (_backend[i] == nullptr) {
                continue;
            }

            if (_backend[i]->init()) {
                detected_num_instances = i+1;
            } else {
                delete _backend[i];
                _backend[i] = nullptr;
            }
        }
    }
```